### PR TITLE
feat: variant on enabled toggle metrics

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

* If toggle is enabled and variant doesn't exist, getVariant should still count enabled toggle as yes
* Unit tests show this new behavior
* This change makes the bahavior same as JS SDK and Java SDK

### CODE REVIEW

File: src/unleash.ts [hunk 1]
```diff
@@ -302,18 +306,21 @@ export class Unleash extends EventEmitter {
...
       this.emit(
         UnleashEvents.Warn,
         `Unleash has not been initialized yet. isEnabled(${name}) defaulted to ${result}`,
       );
     }
+    const { featureEnabled, ...variant } = result;
     if (result.name) {
       this.countVariant(name, result.name);
     }
-    this.count(name, result.enabled);
```
Comment (kwasniew) [inline src/unleash.ts:314]: we were returning variant.enabled not feature enabled

**Review by Tymek — APPROVED** [review]

File: src/test/snapshots/client.test.ts.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ Generated by [AVA](https://avajs.dev).
 
     {
       enabled: true,
+      featureEnabled: true,
```
Comment (gastonfournier) [inline src/test/snapshots/client.test.ts.md:13]: there was a value `enabled` previously. Doesn't it correspond to the feature being enabled? I don't think we have a variant enabled/disabled status

Comment (kwasniew) [reply src/test/snapshots/client.test.ts.md:13]: If that's the case I could massively simplify the fix

Comment (kwasniew) [reply src/test/snapshots/client.test.ts.md:13]: the enabled field is there but I thought it's for the variant not the toggle status

File: src/unleash.ts (hunk 1 above)
Comment (sighphyre) [reply src/unleash.ts:314]: Is this the result of `isEnabled` or is this the raw enabled field that comes back from the Unleash API?

Comment (kwasniew) [reply src/unleash.ts:314]: it wasn't, it was the enabled status of the variant itself.

Comment (kwasniew) [reply src/unleash.ts:314]: In other SDKs we have isEnabled result in scope so we just use it. Here the isEnabled is called 2 layers deep so we need to propagate it up the stack

Comment (kwasniew) [reply src/unleash.ts:314]: This PR is doing the following - for enabled and user activated toggle with no variants:
client.getVariant() returns disabled variant and reports enabled toggle metric

Comment (sighphyre) [reply src/unleash.ts:314]: Thanks! Makes sense!

**Review by sighphyre — APPROVED** [review]
Ye, I think this makes sense

**Pull request merged**